### DISCUSSION TITLE
New User Functions documentation page - feedback welcome!

### DISCUSSION THREAD

**Discussion by hx2A** [py5coding/py5generator#430] [General]
Everyone - I wrote a new documentation page on User Functions:

http://dev.py5coding.org/content/user_functions.html

It covers the new `predraw_update()` function that will be included in the next release, as well as lots of other stuff that probably should have been documented before now.

Any constructive feedback on this new page would be appreciated!

Once this page is squared away I will get ready for the next release, hopefully out by the end of next week.

**Comment by villares** [comment]
Great documentation, thank you!

I have a question about the *Movie Events* section... to use the Processing Movie library, does it need to be "installed", like a jar file downloaded and copied somewhere? If so, then perhaps it should be mentioned.

Something I always think about and I know there is a trade off of verbosity and clarity... should perhaps the first examples be complete with `import py5` and `py5.run_sketch()`? Or, maybe their omission could be mentioned on the text.

About the page structure/ordering, I feel like the  key and mouse events sections are more important than the *Settings Magic* section. So as important as the magic information is, and it is very clearly explained, it kind of eclipses the other sections by preceding them. I'm not sure if it would work but, could the *Settings Magic* become some sort of "extra information box", or maybe put at the end?

**Reply by hx2A** [reply]
Thanks! I just updated the page.

> I have a question about the Movie Events section... to use the Processing Movie library, does it need to be "installed", like a jar file downloaded and copied somewhere? If so, then perhaps it should be mentioned.

Good point. I added that, and also the same for Camera3D.

> should perhaps the first examples be complete with import py5 and py5.run_sketch()? Or, maybe their omission could be mentioned on the text.

I added the import statement to the beginning and `run_sketch()` to each example that users will run.

> I'm not sure if it would work but, could the Settings Magic become some sort of "extra information box", or maybe put at the end?

You are right, it does interrupt the flow of the page a bit, and it beginners reading the page don't need to peek behind the curtain and learn about that. I moved it to the end.
